core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/avro/GenericAvroSerializer.java

```diff
@@ -21,7 +21,7 @@
  * @param <T>
  *            - Base Avro class extended by all generated class files
  */
-public class GenericAvroSerializer<T extends SpecificRecordBase>
+public class GenericAvroSerializer<T>
```

Comment:
Might be a good idea to cache the writers and especially the readers - time to instantiate a  SpecificDatumWriter is >>> time to deserialize an object. 

Here's a similar example I used:
https://github.com/ngageoint/geowave-osm/blob/master/src/main/java/mil/nga/giat/osm/types/TypeUtils.java